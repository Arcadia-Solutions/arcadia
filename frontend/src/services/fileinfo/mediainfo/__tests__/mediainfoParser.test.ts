import { expect, test, describe } from 'vitest'
import MediainfoParser from '../mediainfoParser'

const movMediaInfo = `
General
Complete name                            : Screen Recording 2025-02-12 at 4.04.38â€¯PM.mov
Format                                   : MPEG-4
Format profile                           : QuickTime
Codec ID                                 : qt   0000.00 (qt  )
File size                                : 155 MiB
Duration                                 : 31 s 750 ms
Overall bit rate                         : 41.0 Mb/s
Frame rate                               : 56.277 FPS
Recorded date                            : 2025-02-12T16:04:40-0800
Encoded date                             : 2025-02-13 00:04:41 UTC
Tagged date                              : 2025-02-13 00:05:13 UTC
Writing application                      : macOS 14.7 (23H124)
Writing library                          : Apple QuickTime
Writing hardware                         : Apple Mac15,7

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : Main@L5.1
Format settings                          : CABAC / 2 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 2 frames
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 31 s 750 ms
Source duration                          : 32 s 233 ms
Bit rate                                 : 40.0 Mb/s
Width                                    : 3 456 pixels
Height                                   : 2 234 pixels
Display aspect ratio                     : 3:2
Frame rate mode                          : Variable
Frame rate                               : 56.277 FPS
Minimum frame rate                       : 30.000 FPS
Maximum frame rate                       : 60.000 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0 (Type 1)
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.092
Stream size                              : 152 MiB (98%)
Source stream size                       : 152 MiB (98%)
Title                                    : Core Media Video
Writing library                          : H.264
Encoded date                             : 2025-02-13 00:04:41 UTC
Tagged date                              : 2025-02-13 00:05:13 UTC
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Codec configuration box                  : avcC

Audio
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : 2 / 40 / mp4a-40-2
Duration                                 : 31 s 750 ms
Source duration                          : 31 s 808 ms
Bit rate mode                            : Constant
Bit rate                                 : 128 kb/s
Channel(s)                               : 1 channel
Channel layout                           : M
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 498 KiB (0%)
Source stream size                       : 499 KiB (0%)
Title                                    : Core Media Audio
Encoded date                             : 2025-02-13 00:04:41 UTC
Tagged date                              : 2025-02-13 00:05:13 UTC

`

describe('mediainfoParser', () => {
  test('parses video mediainfo output', () => {
    const parser = new MediainfoParser()
    expect(parser.parse(movMediaInfo)).toMatchSnapshot()
  })
})
