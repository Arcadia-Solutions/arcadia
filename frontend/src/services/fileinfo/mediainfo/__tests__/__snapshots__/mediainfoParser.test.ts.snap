// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mediainfoParser > parses video mediainfo output 1`] = `
{
  "audio": [
    {
      "bit rate": "128 kb/s",
      "bit rate mode": "Constant",
      "channel layout": "M",
      "channel(s)": "1 channel",
      "codec id": "2 / 40 / mp4a-40-2",
      "compression mode": "Lossy",
      "duration": "31 s 750 ms",
      "encoded date": "2025-02-13 00:04:41 UTC",
      "format": "AAC LC",
      "format/info": "Advanced Audio Codec Low Complexity",
      "frame rate": "46.875 FPS (1024 SPF)",
      "id": "2",
      "sampling rate": "48.0 kHz",
      "source duration": "31 s 808 ms",
      "source stream size": "499 KiB (0%)",
      "stream size": "498 KiB (0%)",
      "tagged date": "2025-02-13 00:05:13 UTC",
      "title": "Core Media Audio",
    },
  ],
  "general": {
    "codec id": "qt   0000.00 (qt  )",
    "complete name": "Screen Recording 2025-02-12 at 4.04.38â€¯PM.mov",
    "duration": "31 s 750 ms",
    "encoded date": "2025-02-13 00:04:41 UTC",
    "file size": "155 MiB",
    "format": "MPEG-4",
    "format profile": "QuickTime",
    "frame rate": "56.277 FPS",
    "overall bit rate": "41.0 Mb/s",
    "recorded date": "2025-02-12T16:04:40-0800",
    "tagged date": "2025-02-13 00:05:13 UTC",
    "writing application": "macOS 14.7 (23H124)",
    "writing hardware": "Apple Mac15,7",
    "writing library": "Apple QuickTime",
  },
  "menu": {},
  "text": [],
  "video": [
    {
      "bit depth": "8 bits",
      "bit rate": "40.0 Mb/s",
      "bits/(pixel*frame)": "0.092",
      "chroma subsampling": "4:2:0 (Type 1)",
      "codec configuration box": "avcC",
      "codec id": "avc1",
      "codec id/info": "Advanced Video Coding",
      "color primaries": "BT.709",
      "color range": "Limited",
      "color space": "YUV",
      "display aspect ratio": "3:2",
      "duration": "31 s 750 ms",
      "encoded date": "2025-02-13 00:04:41 UTC",
      "format": "AVC",
      "format profile": "Main@L5.1",
      "format settings": "CABAC / 2 Ref Frames",
      "format settings, cabac": "Yes",
      "format settings, reference frames": "2 frames",
      "format/info": "Advanced Video Codec",
      "frame rate": "56.277 FPS",
      "frame rate mode": "Variable",
      "height": "2 234 pixels",
      "id": "1",
      "matrix coefficients": "BT.709",
      "maximum frame rate": "60.000 FPS",
      "minimum frame rate": "30.000 FPS",
      "scan type": "Progressive",
      "source duration": "32 s 233 ms",
      "source stream size": "152 MiB (98%)",
      "stream size": "152 MiB (98%)",
      "tagged date": "2025-02-13 00:05:13 UTC",
      "title": "Core Media Video",
      "transfer characteristics": "BT.709",
      "width": "3 456 pixels",
      "writing library": "H.264",
    },
  ],
}
`;
